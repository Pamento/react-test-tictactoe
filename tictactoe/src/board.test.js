import React from 'react'
import './setUpTest';
import { shallow, mount } from 'enzyme';
import Board from './Board';

jest.mock('./Square', () => ()=> <div className="square">mockUserCom</div>);

describe('square', () => {
  let squares = Array(9).fill(null);
  it('renders without crashing', () => {
    const wrapper = shallow(<Board squares={squares}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('shows all squares',()=>{
    const wrapper = mount(<Board squares={squares}/>);
    expect(wrapper.find('.square').length).toEqual(9);
  });
})