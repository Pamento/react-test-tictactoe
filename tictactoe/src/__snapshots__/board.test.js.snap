// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`square renders without crashing 1`] = `
<div>
  <div
    className="board-row"
  >
    <Component
      onClick={[Function]}
      value={null}
    />
    <Component
      onClick={[Function]}
      value={null}
    />
    <Component
      onClick={[Function]}
      value={null}
    />
  </div>
  <div
    className="board-row"
  >
    <Component
      onClick={[Function]}
      value={null}
    />
    <Component
      onClick={[Function]}
      value={null}
    />
    <Component
      onClick={[Function]}
      value={null}
    />
  </div>
  <div
    className="board-row"
  >
    <Component
      onClick={[Function]}
      value={null}
    />
    <Component
      onClick={[Function]}
      value={null}
    />
    <Component
      onClick={[Function]}
      value={null}
    />
  </div>
</div>
`;
